Readability.js

```diff
@@ -832,15 +832,18 @@ Readability.prototype = {
     );
 
     // Remove extra paragraphs
+    // At this point, nasty iframes have been removed; only embedded video
+    // ones remain.
+    var contentTags = ["img", "embed", "object", "iframe"];
     this._removeNodes(
       this._getAllNodesWithTag(articleContent, ["p"]),
       function (paragraph) {
-        var imgCount = paragraph.getElementsByTagName("img").length;
-        var embedCount = paragraph.getElementsByTagName("embed").length;
-        var objectCount = paragraph.getElementsByTagName("object").length;
-        // At this point, nasty iframes have been removed, only remain embedded video ones.
-        var iframeCount = paragraph.getElementsByTagName("iframe").length;
-        var totalCount = imgCount + embedCount + objectCount + iframeCount;
+        // We use an array reduction here so the counts of elements are summed
+        // without anyone having to make further code edits if the list of
+        // content tags is changed.
+        var totalCount = contentTags.reduce(function (total, tag) {
+          return total + paragraph.getElementsByTagName(tag).length;
+        }, 0);
```

Comment:
So I'm a little lost. Any reason not to use `this._getAllNodesWithTag(paragraph, contentTags).length` ? That would seem easier than hand-rolling the reduce.

The other option, for reasons of speed, would be to iterate over `contentTags` (without a nested function) and to return `false` as soon as we find any of the content tags (as the `return` below will be `false` as soon as we find any of the elements in question). It might also make sense to put the `_getInnerText` call first in that case.

Both of those would still have the same benefit in terms of not requiring manual copy-pasting for each kind of inner tag that we're looking for.

Reply:
I see what you mean: I fixed one problem of weird methodology in the original, but not the other. I'll iterate further when I find some time.

Reply:
@JKingweb any chance you had a moment to look at this again? :-)